src/lex/cpp.rs

```diff
@@ -923,16 +923,34 @@ impl<'a> PreProcessor<'a> {
                 Vec::new()
             };
             let body = body(self)?;
-            self.definitions
-                .insert(id.data, Definition::Function { params, body });
+            self.define_macro(id.data, Definition::Function { params, body })
+                .map_err(|e| self.span(start).with(e))?;
             Ok(())
         } else {
             // object macro
             let tokens = body(self)?;
-            self.definitions.insert(id.data, Definition::Object(tokens));
+            self.define_macro(id.data, Definition::Object(tokens))
+                .map_err(|e| self.span(start).with(e))?;
             Ok(())
         }
     }
+    fn define_macro(&mut self, name: InternedStr, definition: Definition) -> Result<(), CppError> {
+        use std::collections::hash_map::Entry;
+        match self.definitions.entry(name) {
+            Entry::Vacant(entry) => {
+                entry.insert(definition);
+                Ok(())
+            }
+            Entry::Occupied(entry) => {
+                // https://port70.net/~nsz/c/c11/n1570.html#6.10.3p1
```

Comment:
Let's just put the section and not the full URL.
```suggestion
                // 6.10.3p1
```

Reply:
Look 9 lines down ;)